src/Silverback.Core/Util/ReflectionHelper.cs

```diff
@@ -6,10 +6,9 @@
 
 namespace Silverback.Util
 {
-    // TODO: Test
     internal static class ReflectionHelper
     {
-        public static bool IsAsync(this MethodInfo methodInfo) =>
```

Comment:
To check if a method is async, one would need to check for an compiler generated attribute.